src/main/java/Factory.java

```diff
@@ -0,0 +1,3 @@
+public interface Factory {
```

Comment:
Let's rename to 'BookSpecificationFactory' because we're using 'BookFieldType' + 'Specification<T>' types here